tee_supplicant/src/tee_supplicant.c

```diff
@@ -120,6 +131,519 @@ static int load_ta(uint32_t num_params, struct tee_param *params)
 	return TEEC_SUCCESS;
 }
 
+static int tee_fs_open(size_t num_params, struct tee_param *params, int flags)
+{
+	struct tee_shm *shm;
+	char *name, path[REE_FS_PATH_MAX] = REE_FS_MP;
+	int fd;
+
+	if (num_params != 3) {
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	if ((params[0].attr & TEE_PARAM_ATTR_TYPE_MASK) !=
+		TEE_PARAM_ATTR_TYPE_VALUE_INPUT ||
+	    (params[1].attr & TEE_PARAM_ATTR_TYPE_MASK) !=
+		TEE_PARAM_ATTR_TYPE_MEMREF_INPUT ||
+	    (params[2].attr & TEE_PARAM_ATTR_TYPE_MASK) !=
+		TEE_PARAM_ATTR_TYPE_VALUE_OUTPUT)
+	{
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	/*TODO do all safety checks */
+	shm = (struct tee_shm *)params[1].c;
+
+	if (!shm || !shm->addr) {
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	name = shm->addr;
+	strncat(path, name, PATH_MAX);
+	fd = open(path, flags, 0600);
+	if (fd < 0) {
+		int rc = -errno;
+		if (rc == -ENOENT) {
+			return TEEC_ERROR_ITEM_NOT_FOUND;
+		}
+		LOG_ERR("failed to open/create %s (%d)", path, rc);
+		return TEEC_ERROR_GENERIC;
+	}
+
+	params[2].a = fd;
+
+	return TEEC_SUCCESS;
+}
+
+static int tee_fs_close(size_t num_params, struct tee_param *params)
+{
+	if (num_params != 1 || (params[0].attr & TEE_PARAM_ATTR_TYPE_MASK) !=
+		TEE_PARAM_ATTR_TYPE_VALUE_INPUT)
+	{
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	if (close((int)params[0].b) < 0) {
```

Comment:
I suggest something like this:
```
int fd=(int)params[0].b;
if (close(fd < 0)) {
  ...
```
It will increase readability

Reply:
Also, cast to `int` is not needed

Reply:
done